agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/processors/AgentProcessor.java

```diff
@@ -61,14 +84,26 @@ public static StrictIncludeExclude create(ProcessorIncludeExclude includeExclude
             if (spanNames == null) {
                 spanNames = new ArrayList<>();
             }
-            return new StrictIncludeExclude(attributes, spanNames);
+            List<String> logNames = includeExclude.logNames;
+            if (logNames == null) {
+                logNames = new ArrayList<>();
+            }
+            return new StrictIncludeExclude(attributes, spanNames, logNames);
         }
 
-        // Function to compare span with user provided span names
-        public boolean isMatch(SpanData span) {
-            if (!spanNames.isEmpty() && !spanNames.contains(span.getName())) {
-                // span name doesn't match
-                return false;
+        // Function to compare span with user provided span names and log names
+        public boolean isMatch(SpanData span, boolean isLog) {
+
+            if (spanNames.isEmpty() && logNames.isEmpty()) {
+                // check attributes for both spans and logs
+                return this.checkAttributes(span);
+            }
+            if (isLog) {
+                if (logNames.isEmpty()) return false;
+                if (!logNames.isEmpty() && !logNames.contains(span.getName())) return false;
+            } else {
+                if (spanNames.isEmpty()) return false;
+                if (!spanNames.isEmpty() && !spanNames.contains(span.getName())) return false;
             }
             return this.checkAttributes(span);
```

Comment:
this method is a bit hard to follow, I think this makes it clearer, but is also slightly different, wdyt?
```suggestion
            // check spanNames and logNames
            if (isLog) {
                if (!spanNames.isEmpty()) {
                    return false;
                }
                if (!logNames.isEmpty() && !logNames.contains(span.getName())) {
                    return false;
                }
            } else {
                if (!logNames.isEmpty()) {
                    return false;
                }
                if (!spanNames.isEmpty() && !spanNames.contains(span.getName())) {
                    return false;
                }
            }
            // check attributes
            return this.checkAttributes(span);
```

Reply:
@trask this logic will not work for cases where user provides both spanNames and logNames in the processor.

Reply:
what does it mean if both are included? should that be valid?